app/javascript/dashboard/components/widgets/conversation/helpers/SLAHelper.js

```diff
@@ -0,0 +1,135 @@
+const calculateThreshold = (timeOffset, threshold) => {
+  // Calculate the time left for the SLA to breach or the time since the SLA has breached
+  if (threshold === null) return null;
+  const currentTime = Math.floor(Date.now() / 1000);
+  return timeOffset + threshold - currentTime;
+};
+
+const findMostUrgentSLAStatus = SLAStatuses => {
+  // Sort the SLAs based on the threshold and return the most urgent SLA
+  SLAStatuses.sort((a, b) => Math.abs(a.threshold) - Math.abs(b.threshold));
+  return SLAStatuses[0];
+};
+
+const convertToMinutes = time => Math.floor(time / 60);
+const convertToHoursAndMinutes = minutes => [
+  Math.floor(minutes / 60),
+  minutes % 60,
+];
+const convertToDaysAndHours = hours => [Math.floor(hours / 24), hours % 24];
+const convertToMonthsAndDays = days => [Math.floor(days / 30), days % 30];
+const convertToYearsAndMonths = months => [
+  Math.floor(months / 12),
+  months % 12,
+];
+
+const formatTime = (value, label, nextValue, nextLabel) => {
+  if (value > 0) return `${value}${label} ${nextValue}${nextLabel}`;
+  return null;
+};
+
+const formatDuration = (years, months, days, hours, minutes) => {
+  return (
+    formatTime(years, 'y', months, 'mo') ||
+    formatTime(months, 'mo', days, 'd') ||
+    formatTime(days, 'd', hours, 'h') ||
+    formatTime(hours, 'h', minutes, 'm') ||
+    `${minutes > 0 ? minutes : 1}m`
+  );
+};
+
+// Format the SLA time in the format of years, months, days, hours and minutes, eg. 1y 2mo 3d 4h 5m
+export const formatSLATime = time => {
+  if (time < 0) return '';
+
+  const minutes = convertToMinutes(time);
```

Comment:
```suggestion
export const formatSLATime = seconds => {
  if (seconds < 0) return '';

  const minutes = convertToMinutes(seconds);
```

just improving variable names